runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/SimpleParDoFn.java

```diff
@@ -397,6 +397,9 @@ private void processSystemTimer(TimerData timer) throws Exception {
           window,
           targetTime);
 
+      fnRunner.onWindowExpiration(
+          window, timer.getOutputTimestamp(), this.stepContext.stateInternals().getKey());
```

Comment:
@reuvenlax, Can you guide why we are not considering outputTimestamp in WindmillTimerInternals.windmillTimerToTimerData. TimerData is created as follows:
`TimerData.of(id, familyId, namespace, timestamp, timerTypeToTimeDomain(timer.getType()));

I think this is where we lost outputTimestamp.

Reply:
Why do we need the output timestamp in onWindowExpiration?

Reply:
I think elements that are output on OnWindowExpiration would have a timestamp equals the output timestamp of the GC timer that's why we need it here. In the case of a window of Instant [0-9], 10 would be GC timer with an output timestamp of 9. So elements that are output from onWindowExpiration would have a timestamp of 9. @reuvenlax